<a href="https://colab.research.google.com/github/Emenike-Amara/Projects/blob/main/Customers_Purchase_Inisghts_Unlocking_Targeted_Marketing_Opportunities.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ℹ Promblem Statement


> Understanding customer behavior is crucial for business success. Targeted marketing, which involves tailoring marketing efforts to specific customer segments, has emerged as an effective strategy to drive sales and enhance customer satisfaction. However, many businesses struggle to unlock targeted marketing opportunities due to a lack of customer purchase insights. This lack of a comprehensive and holistic understanding of customer purchasing behavior hinders business' ability to create personalized marketing campaigns that resonate with customers and drive desired outcomes.

> Hence, there is a pressing need for companies to unlock targeted marketing opportunities by leveraging customer purchase insights effectively. By doing so, companies can enhance customer engagement, increase sales, and gain a competitive advantage in the market.








# ℹ Background Statement : [Dataset](https://www.kaggle.com/datasets/aungpyaeap/supermarket-sales) credit -- Kaggle.



> The decision to focus on the dataset of supermarkets is because of the recorded growth of supermarkets are increasing and market competitions are high. This truth holds universal relevance in today's dynamic business landscape. The insights gained from this dataset can be applied to any industry, opening up exciting opportunities for businesses across different industries. By leveraging the power of these insights, companies can stay ahead of the competition, drive sales, and enhance customer satisfaction in their respective industries. The potential for innovation and success is immense, making this choice of dataset truly exciting and promising for businesses looking to unlock new possibilities in their marketing strategies.


# ▶ Data Cleaning Process
> In Excel: Format the data column

> In PowerBI: Introduced a new measure - Total Orders
New column = DayofWeek (Sunday, Monday..., Saturday)
New column = Promoter grouping 
New Measure = Net Promoter Score



# ▶ Load DataSet for exploration 

Installing the Powerbiclient Python package, this package allows the embedding of a PowerBI report in GC easily.

In [ ]:
pip install powerbiclient

In [2]:
#Load the necessary Libraries
from powerbiclient import Report, models
from powerbiclient.authentication import DeviceCodeLoginAuthentication
import pandas as pd
from google.colab import drive
from google.colab import output
from urllib import request 
import requests

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [4]:
#The dataset here has formatted Date
path='/content/drive/My Drive/supermarket_sales.xlsx'
data = pd.read_excel(path)

In [5]:
data.head()

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%  Total Purchase  ...  \
0       Health and beauty       74.69         7  26.1415          522.83  ...   
1  Electronic accessories       15.28         5   3.8200           76.40  ...   
2      Home and lifestyle       46.33         7  16.2155          324.31  ...   
3       Health and beauty       58.22         8  23.2880          465.76  ...   
4       Sports and travel       86.31         7  30.2085          604.17  ...   

   Day  Year       Date  Week      Time  Paymenttype    cogs  \
0    5  2019   5/1/2019     1  13:08:00      Ewallet  522.83   
1    8  2019   8/3/2019    10  10:29:00         Cash   76.40   
2    3  2019   3/3/2019     9  13:23:00  Credit card  324.31   
3   27  2019  27/1/2019     4  20:33:00      Ewallet  465.76   
4    8  2019   8/2/2019     6  10:37:00      Ewallet  604.17   

  gross margin percentage gross income  Rating  
0                4.761905      26.1415     9.1  
1                4.761905       3.8200     9.6  
2                4.761905      16.2155     7.4  
3                4.761905      23.2880     8.4  
4                4.761905      30.2085     5.3  

[5 rows x 23 columns]

In [6]:
data.describe()

Unit price     Quantity       Tax 5%  Total Purchase   Total Cost  \
count  1000.000000  1000.000000  1000.000000      999.000000  1000.000000   
mean     55.672130     5.510000    15.379369      307.276276   322.348369   
std      26.494628     2.923431    11.708825      234.086958   245.841398   
min      10.080000     1.000000     0.508500       10.170000    10.678500   
25%      32.875000     3.000000     5.924875      117.895000   122.511375   
50%      55.230000     5.000000    12.088000      241.600000   253.596000   
75%      77.935000     8.000000    22.445250      448.580000   470.998500   
max      99.960000    10.000000    49.650000      993.000000  1042.650000   

             Total        Month          Day    Year         Week        cogs  \
count  1000.000000  1000.000000  1000.000000  1000.0  1000.000000  1000.00000   
mean    322.967430     1.993000    15.256000  2019.0     6.996000   307.58738   
std     245.885557     0.835254     8.693563     0.0     3.596515   234.17651   
min      10.680000     1.000000     1.000000  2019.0     1.000000    10.17000   
25%     124.425000     1.000000     8.000000  2019.0     4.000000   118.49750   
50%     253.850000     2.000000    15.000000  2019.0     7.000000   241.76000   
75%     471.350000     3.000000    23.000000  2019.0    10.000000   448.90500   
max    1042.650000     3.000000    31.000000  2019.0    13.000000   993.00000   

       gross margin percentage  gross income      Rating  
count              1000.000000   1000.000000  1000.00000  
mean                  4.761905     15.379369     6.97270  
std                   0.000000     11.708825     1.71858  
min                   4.761905      0.508500     4.00000  
25%                   4.761905      5.924875     5.50000  
50%                   4.761905     12.088000     7.00000  
75%                   4.761905     22.445250     8.50000  
max                   4.761905     49.650000    10.00000

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 23 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total Purchase           999 non-null    float64
 10  Total Cost               1000 non-null   float64
 11  Total                    1000 non-null   float64
 12  Month                    1000 non-null   int64  
 13  Day                      1000 non-null   int64  
 14  Year                     

# ▶ Introducing the already designed PowerBi report in here

In [8]:
#Import the DeviceCodeLoginAuthentication class to authenticate against PowerBI and initiate the Micrcosoft device authentication

device_auth = DeviceCodeLoginAuthentication()

group_id = "090a00bd-5bdb-4739-97ba-d8f0e222460d"
report_id = "b7c96304-697e-49d5-a13f-ed654bf100e6"

report = Report(group_id= group_id, report_id= report_id, auth= device_auth)


Performing device flow authentication. Please follow the instructions below.
To sign in, use a web browser to open the page https://microsoft.com/devicelogin and enter the code EDPCCHWWK to authenticate.

Device flow authentication successfully completed.
You are now logged in.


In [9]:
report.set_size (800,906)

output.enable_custom_widget_manager()

In [10]:
report

Report(container_height=800.0, container_width=906.0)

** ℹ View the [report](https://www.novypro.com/project/portfolio-10)** here

# ℹ Interpretation & Actionable Insights
Income was considered for this report, since the pivotal goal of the business is to make profit. Here some insights worthy to note:

> The analysis of the Home & Lifestyle product line reveals higher purchases from females overall, with Mandalay being the best-performing location. However, our Net Promoter Score (NPS) is at ***-4.76%***, indicating **low** customer loyalty, satisfaction, and enthusiasm. NPS is measured by asking customers to rate their likelihood of recommending our product/company. To improve customer satisfaction and loyalty, we need to engage with customers to better understand their experiences and expectations, based on the current NPS score.

> The analysis of our Health & Beauty product line reveals that males are the dominant gender,which is an intriguing finding that may indicate a potential gap in our inventory of female-centric products. This insight prompts us to reflect on the importance of considering gender diversity and inclusivity in our product offerings. By adopting a holistic and inclusive analytical approach, we can better understand and cater to the diverse needs of all customers, driving business success and fostering a more customer-centric approach in our health & beauty product line.

> Food & Beverage ads should be targeted to Females in Naypytaw.

> Sports & Travel ads should be targeted to Males in Mandalay & Yangon

> Myannar has NPS Score of **-4.22%** this is an indication that this branch has unhappy customers who are unlikely repeat purchase, and may even discourage others from buying purchasing.

> In Yangon has a NPS Score of **1.76%**, we have more Passive customers who can lean towards detractors if they go through bad experiences and promoters if they get positive experiences. It is quite clear the need to convert to promoters, its easier to convert them to promoters than detractors to promoters. There is a need to understand their experience, iterate this sucess and drive increased revenue


